db/109_migrate.sql

```diff
@@ -0,0 +1,144 @@
+ TABLE_NAME='majormodversioncached') ) THEN
+ START TRANSACTION;
+    ALTER TABLE `majormodversioncached` ADD COLUMN `majorGameVersion` BIGINT UNSIGNED NOT NULL;
+    UPDATE `majormodversioncached` mmvc
+      INNER JOIN `majorversion` mv on mv.majorversionid = mmvc.majorversionid
+      SET `majorGameVersion` = compile_semantic_version(REPLACE(mv.name, 'x', '0') & 0xffffffff00000000);
+
+    ALTER TABLE `majormodversioncached` DROP INDEX `majorversionid`;
+    ALTER TABLE `majormodversioncached` DROP INDEX `index`;
+    ALTER TABLE `majormodversioncached` DROP COLUMN `majorversionid`;
+
+    ALTER TABLE `majormodversioncached` DROP INDEX `modid`;
+    ALTER TABLE `majormodversioncached` CHANGE COLUMN `modid` `modId` INT NOT NULL;
+    ALTER TABLE `majormodversioncached` ADD PRIMARY KEY (`modId`, `majorGameVersion`);
+
+    ALTER TABLE `majormodversioncached` RENAME TO `ModCompatibleMajorGameVersionsCached`;
+    
+    DROP TABLE IF EXISTS `majorversion`;
+  COMMIT;
+END IF;
+
+
+IF EXISTS( (SELECT 1 FROM `tag` WHERE `tagtypeid` = 1) ) THEN
+ START TRANSACTION;
+
+  INSERT INTO `ModReleaseCompatibleGameVersions` (`releaseId`, `gameVersion`)
+    SELECT r.releaseid, compile_semantic_version(t.name)
+    FROM `assettag` AS ast
+    JOIN `asset` a ON a.assetid = ast.assetid AND a.assettypeid = 2 -- release
+    JOIN `release` r ON r.assetid = a.assetid
+    JOIN `tag` t ON t.tagid = ast.tagid
+    WHERE t.tagtypeid = 1;
+
+  DELETE FROM `tag` WHERE `tagtypeid` = 1; -- gameversion
+  DELETE FROM `tagtype` WHERE `tagtypeid` = 1; -- gameversion
+  
+  COMMIT;
+END IF;
+
+IF EXISTS( (SELECT * FROM information_schema.COLUMNS WHERE TABLE_SCHEMA='moddb' AND
+ TABLE_NAME='release' AND COLUMN_NAME='modversion' AND DATA_TYPE='varchar') ) THEN
+ START TRANSACTION;
+    ALTER TABLE `release` CHANGE COLUMN `modversion` `_modversion` VARCHAR(50) NULL;
+
+    ALTER TABLE `release` ADD COLUMN `modversion` BIGINT UNSIGNED NOT NULL AFTER `modidstr`;
+    UPDATE `release` SET `modversion` = compile_semantic_version(`_modversion`);
+
+    ALTER TABLE `release` DROP INDEX `modidstr`;
+    ALTER TABLE `release` ADD UNIQUE INDEX `identifier` (`modId`, `modversion`);
+
+    ALTER TABLE `release` DROP COLUMN `_modversion`;
+  COMMIT;
+END IF;
+
+IF EXISTS( (SELECT * FROM information_schema.COLUMNS WHERE TABLE_SCHEMA='moddb' AND
+ TABLE_NAME='modpeek_result' AND COLUMN_NAME='detectedmodversion' AND DATA_TYPE='varchar') ) THEN
+ START TRANSACTION;
+    ALTER TABLE `modpeek_result` CHANGE COLUMN `detectedmodversion` `_detectedmodversion` VARCHAR(255);
+
+    ALTER TABLE `modpeek_result` ADD COLUMN `detectedmodversion` BIGINT UNSIGNED NOT NULL AFTER `detectedmodidstr`;
+    UPDATE `modpeek_result` SET `detectedmodversion` = compile_semantic_version(`_detectedmodversion`);
+
+    ALTER TABLE `modpeek_result` DROP COLUMN `_detectedmodversion`;
+  COMMIT;
+END IF;
+
+
+END $$
+
+CALL upgrade_database__moderation() $$
+
+DROP FUNCTION compile_semantic_version $$
+
+DELIMITER ;
```

Comment:
Likely already part of your deploy plan but please ensure this is tested on backup data first and make a clean backup before running this.